### ISSUE TITLE
Katana: PointInstancer Op produces recursive hierarchy in some cases

### ISSUE THREAD

**Issue by nrusch** [PixarAnimationStudios/USD#286]
### Description of Issue

With certain stage configurations, the Katana reader Op for `PointInstancer` prims will generate an infinitely recursive scene graph hierarchy below the `"usd point instancer"` location, and will not generate `"instance source"` locations for the prototype prims.

To reproduce, simply read the USD file in the attached [PointInstancerTest.zip](https://github.com/PixarAnimationStudios/USD/files/1307954/PointInstancerTest.zip) into Katana, and expand the hierarchy under the `USD_TEST_INSTANCER ` location.

If the `material:binding` attribute values are removed from the prototype source prims, the stage seems to load as expected.

### Package Versions

USD 0.8.0

**Comment by sirpalee** [comment]
I was digging around a bit and found why the infinite recursion happens.

The example scene looks like this. The materials placed into their own separate scope by the Arnold shader exporter from usd-arnold.

```
-root
--- xform
----- instancer
--- xform
----- instances
--- materials
```

When the instancer exporter runs, it tries to build both the instance sources and the assigned materials by looking for a common prefix for both. Since the common prefix is above the instancer in our case, the instancer builder triggers its export once again.

A quick patch like this gets rid of the infinite recursion; it's just quickly checking if the common prefix is above the instancer. As expected, this breaks all the material bindings.

```
diff --git a/third_party/katana/lib/usdKatana/readPointInstancer.cpp b/third_party/katana/lib/usdKatana/readPointInstancer.cpp
index 7a9dbfa8..4b93f71e 100644
--- a/third_party/katana/lib/usdKatana/readPointInstancer.cpp
+++ b/third_party/katana/lib/usdKatana/readPointInstancer.cpp
@@ -361,8 +361,8 @@ PxrUsdKatanaReadPointInstancer(
     //
     // Validate instancer data.
     //
-
-    const std::string instancerPath = instancer.GetPath().GetString();
+    const auto instancerSdfPath = instancer.GetPath();
+    const std::string instancerPath = instancerSdfPath.GetString();

     UsdStageWeakPtr stage = instancer.GetPrim().GetStage();

@@ -568,7 +568,7 @@ PxrUsdKatanaReadPointInstancer(
                 {
                     const SdfPath &commonPrefix =
                             protoPath.GetCommonPrefix(materialPath);
-                    if (commonPrefix.GetString() == "/")
+                    if (commonPrefix.GetString() == "/" || instancerSdfPath.HasPrefix(commonPrefix))
                     {
                         // XXX Unhandled case.
                         // The prototype prim and its material are not under the
```

**Comment by sirpalee** [comment]
The material issues are the following. Based on the comments, it seems to be a known case that pxrUsdIn is not handling. Any pointers to fix it would be greatly appreciated.

```
UsdIn: Failed to compute katana path for usd path: /fxUSDInstancer_world/Looks with given isolatePath: /fxUSDInstancer_world/fxUSDInstancer_world/pPyramid1
UsdIn: Failed to compute katana path for usd path: /fxUSDInstancer_world/Looks with given isolatePath: /fxUSDInstancer_world/fxUSDInstancer_world/pCube1
UsdIn: Failed to compute katana path for usd path: /fxUSDInstancer_world/Looks with given isolatePath: /fxUSDInstancer_world/fxUSDInstancer_world/pTorus1
UsdIn: Failed to compute katana path for usd path: /fxUSDInstancer_world/Looks with given isolatePath: /fxUSDInstancer_world/fxUSDInstancer_world/pHelix1
```

**Comment by jtran56** [comment]
Filed as internal issue #150968.

**Comment by sunyab** [comment]
The fix from pull request #384 was released with v0.8.4. Closing, thanks!
